### PR TITLE
Reader: Pull suggestions from subscribed tags, use what we have

### PR DESCRIPTION
**Pull request by blowery**
Bump the suggestion provider out to a separate HoC and use it from search and following.

Enhancement to #11643

### CODE REVIEW

File: client/reader/search-stream/suggestion-provider.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+		componentWillMount() {
+			TagSubscriptions.on( 'change', this.handleChange );
+		}
+
+		componentWillUnmount() {
+			TagSubscriptions.off( 'change', this.handleChange );
+		}
+
+		suggestionsFromTags() {
```
Comment (samouri) [inline client/reader/search-stream/suggestion-provider.jsx]: what if you follow less than 3 tags?

Comment (blowery) [inline client/reader/search-stream/suggestion-provider.jsx]: good call. how many tags do you think we should require before using them as a source?

or maybe we should mix tags and the hard-wired suggestions?

**Comment by samouri** [comment]
Is there any reason to use a HoC versus some function inside of `search-stream/suggestions.js`: `getSuggestions()`?

**Comment by blowery** [comment]
> Is there any reason to use a HoC versus some function inside of search-stream/suggestions.js: getSuggestions()?

The HoC's sole purpose in life is to react to changes in the tag subscription store.

File: client/reader/search-stream/suggestion-provider.jsx (hunk 1 above)
Comment (samouri) [inline client/reader/search-stream/suggestion-provider.jsx]: cc @fraying 

> or maybe we should mix tags and the hard-wired suggestions?
makes sense to me!

**Comment by fraying** [comment]
We should prefer user-entered data. My bet is it will perform better than general topics.

**Comment by blowery** [comment]
rejiggered this a bit to avoid the switch. We lose tags on the initial load, but get them if you leave and come back, or if you don't start with following. 

I tried delaying showing suggestions until the tag store loads, but that causes the list of cards to bump down once the tags came in. That felt even worse than the flicker.

**Comment by fraying** [comment]
Kind of defeats the point if it's not there on first load, since that may be our one and only chance for most members. Any other solutions?

**Comment by samouri** [comment]
@fraying, the work I'm currently doing in https://github.com/Automattic/wp-calypso/pull/11627 will fix it (reduxifying tags)

**Comment by blowery** [comment]
> Any other solutions?

1) delaying the appearance of suggestions and eating the vertical shift
2) holding space for the suggestions and maybe eating a shift up if there are no suggestions. there would be no shift for EN folks, but there would for non-EN / less than 3 tags folks
3) delaying the entire render of the following and search pages until tags have loaded

**Comment by blowery** [comment]
> the work I'm currently doing in #11627 will fix it (reduxifying tags)

that'll only fix it for subsequent visits

> Kind of defeats the point if it's not there on first load, since that may be our one and only chance for most members

Most members don't subscribe to any tags, so I don't know if that's really true

**Comment by fraying** [comment]
> Most members don't subscribe to any tags, so I don't know if that's really true

Ok, trying again: Of the people who follow tags, it may be our one and only chance. In other words, they'll glance at the thing at the top once in the beginning. It's much harder to get them to look again later.

**Comment by fraying** [comment]
I like this idea:

> 2. holding space for the suggestions and maybe eating a shift up if there are no suggestions. there would be no shift for EN folks, but there would for non-EN / less than 3 tags folks

What about holding the space and not shifting either way? That way, the space is filled for ppl with followed tags and EN peeps, and for non-EN/no followed tags, they just get some extra whitespace, no shift.

Thoughts?

**Comment by blowery** [comment]
Trying out 2 in #11683

**Comment by samouri** [comment]
@fraying, I think the current approach is reasonable because if we split our users into three camps:
1. users that don't follow tags: n/a
2. users that follow tags AND only visit calypso once ever: does this category even exist? If it does then they will see our default suggestions instead of their followed tags.
3. users that follow tags and periodically/regularly visit Calypso: they will see their tags as we intend and hopefully it'll capture their attention during a drive by

**Comment by fraying** [comment]
>So I'm not sure which user is missing out on an optimal ux here

Users that follow tags and don't see them on their first visit. They're likely to ignore the whole area on subsequent visits, since they'll assume they're unpersonalized.

**Comment by blowery** [comment]
killed in favor of #11683

**Pull request closed without merging**